src/lib/containers/widgets/RangeSlider.tsx

```diff
@@ -0,0 +1,225 @@
+import * as React from 'react'
+import './RangeSlider.scss'
+
+import { Slider, Rail, Handles, Tracks, Ticks } from 'react-compound-slider'
+import {
+  SliderItem,
+  GetHandleProps,
+  GetTrackProps,
+} from 'react-compound-slider'
+import { RangeValues } from './Range'
+import { useState } from 'react'
+
+
+/**************** RANGE SLIDER SUBCOMPONENTS *********************/
+
+/***  handles ***/
+
+interface IHandleProps {
+  domain: number[]
+  handle: SliderItem
+  getHandleProps: GetHandleProps
+}
+
+export const Handle: React.FunctionComponent<IHandleProps> = ({
+  domain: [min, max],
+  handle: { id, value, percent },
+  getHandleProps,
+}) => (
+  <div
+    role="slider"
+    className="rangeSlider__handle"
+    aria-valuemin={min}
+    aria-valuemax={max}
+    aria-valuenow={value}
+    style={{
+      left: `${percent}%`,
+    }}
+    {...getHandleProps(id)}
+  />
+)
+
+
+/***   track  ***/
+interface ITrackProps {
+  source: SliderItem
+  target: SliderItem
+  getTrackProps: GetTrackProps
+}
+
+export const Track: React.FunctionComponent<ITrackProps> = ({
+  source,
+  target,
+  getTrackProps,
+}) => (
+  <div
+    className="rangeSlider__track"
+    style={{
+      left: `${source.percent}%`,
+      width: `${target.percent - source.percent}%`,
+    }}
+    {...getTrackProps()}
+  />
+)
+
+/***   tick  ***/
+interface ITickProps {
+  key: string
+  tick: SliderItem
+  count: number
+  mode?: 'SHOWALL' | 'SHOWNONE' | 'SHOWMINMAX'
+}
+
+export const Tick: React.FunctionComponent<ITickProps> = ({
+  tick,
+  count,
+  mode = 'SHOWMINMAX',
+}) => (
+  <div>
+    <div
+      className="rangeSlider__tick"
+      style={{
+        left: `${tick.percent}%`,
+      }}
+    />
+    <div
+      className="rangeSlider__tickInner"
+      style={{
+        marginLeft: `${-(100 / count) / 2}%`,
+        width: `${100 / count}%`,
+        left: `${tick.percent}%`,
+      }}
+    >
+      {(mode === 'SHOWALL' ||
+        (mode === 'SHOWMINMAX' &&
+          (tick.percent == 0 || tick.percent == 100))) &&
+        tick.value}
+    </div>
+  </div>
+)
+
+
+/*************  RANGE SLIDER COMPONENT ****************/
+
+
+export type RangeSliderProps = {
+  domain: string[]
+  initialValues: RangeValues
+  step: number
+  doUpdateOnApply?: boolean
+  maxTickCount?: number
+  onChange: Function
+}
+
+export type RangeSliderState = {
+  values: readonly number[]
+}
+
+function getInitialValues(initialValues: RangeValues, domain: string[]) {
+  const result = [
+    initialValues.min ? +initialValues.min : +domain[0],
```

Comment:
I've never seen this syntax before, does this add one to the values?

Reply:
It casts string to a number.

Reply:
I see, that's neat, I think it would be easier to read the code if those were cast using Number(val)